jupyter_server/services/kernels/kernelmanager.py

```diff
@@ -532,6 +532,24 @@ def _check_kernel_id(self, kernel_id):
             raise web.HTTPError(404, "Kernel does not exist: %s" % kernel_id)
 
     # monitoring activity:
+    tracked_message_types = List(
+        trait=Unicode(),
+        config=True,
+        default_value=[
+            "comm_close",
+            "comm_msg",
+            "comm_open",
+            "complete_request",
+            "is_complete_request",
+            "execute_input",
+            "execute_reply",
+            "execute_request",
+            "inspect_request",
```

Comment:
```suggestion
            "inspect_request",
            "debug_request",
            "debug_reply",
```


https://jupyter-client.readthedocs.io/en/stable/messaging.html#debug-request

Maybe also `history_request`/`history_reply`

Reply:
I see it was also discussed in depth in https://github.com/jupyter-server/jupyter_server/issues/1360#issuecomment-1816384824. I am not sure if there was a resolution though.

Reply:
> I am not sure if there was a resolution though.

We didn't come to a consensus on the set of message types to include, but I think we *did* get a pretty clear consensus that the set of message types should be configurable so that server admins can override whatever default we choose.

I don't know if we can get a consensus on the set of types to leave in the default value for the config, but if we do get a consensus that something should be added to (or removed from) this list, then please let me know.

Reply:
As discussed on the call today, it might be more conservative to change this list to a `not_tracked_message_types`, as it would be less risk of accidentally forgetting that a certain message type should be counted as kernel activity. It would also mean that the risk changes from "culling a kernel that shouldn't be culled" (bad because of potential data loss) to "not culling a kernel that can be culled" (bad because of potential cost / resource use), where the former seems more disruptive.

Reply:
I've put together a tentative list of what I *think* would be the default set of not-tracked message types.

I got this by going through [this page](https://jupyter-client.readthedocs.io/en/latest/messaging.html#messaging-in-jupyter), and selecting all of the message types that were either:

1. Not sent on the shell channel, or
2. Of the form `*_info_(request|reply)`.

This is *almost* disjoint from the proposed list for tracked message types. The difference is that the `execute_input` message that was going to be tracked is included here in the not-tracked list because it is sent on the IOPub channel instead of the shell channel.

The full list is:

`comm_info_request`,
`comm_info_reply`,
`kernel_info_request`,
`kernel_info_reply`,
`shutdown_request`,
`shutdown_reply`,
`interrupt_request`,
`interrupt_reply`,
`debug_request`,
`debug_reply`,
`stream`,
`display_data`,
`update_display_data`,
`execute_input`,
`execute_result`,
`error`,
`status`,
`clear_output`,
`debug_event`,
`input_request`,
`input_reply`

@krassowski @vidartf @Zsailer Can you all please help me double check this?

In particular:

1. Is my methodology described above the right approach to take?
2. Does the list I came up with look right?

Thanks, in advance!

Reply:
Thanks @ojarjur. I'll give this a closer look later today.

Reply:
This list looks right to me.

Thanks, @ojarjur. I believe this is good to go.